atc/exec/retry_step_test.go

```diff
@@ -339,4 +339,16 @@ var _ = Describe("Retry Step", func() {
 			})
 		})
 	})
+
+	Context("when the RetryStep is given no attempts", func() {
+		BeforeEach(func() {
+			step = Retry()
+		})
+
+		Describe("Succeeded", func() {
+			It("should return false", func() {
+				Expect(step.Succeeded()).To(BeFalse())
+			})
+		})
+	})
```

Comment:
Could we add another similar case or replace this case, where if `Retry.Succeeded` is invoked with attempts>0, but without first invoking `Run`, it should also return false ?

Reply:
thanks, we added this test.